### PR TITLE
DEV-4178: Queries and tests for annotation and object detection metrics

### PR DESCRIPTION
**Pull request by jkgenser**
Also adds a client test for creating an annotation model

### CODE REVIEW

**Comment by jkgenser** [comment]
Tests pass against dev.

File: indico/queries/metrics.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
+from indico.client.request import GraphQLRequest
```
Comment (nkassis) [inline indico/queries/metrics.py:1]: I think this probably should be a submodule of model_groups to make it clear what these are.

File: indico/queries/metrics.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+from indico.types.metrics import SequenceMetrics
+import json
+
```
Comment (nkassis) [inline indico/queries/metrics.py:4]: We can probably add a GetModelMetrics query here that first gets the model group and type the gets the appropriate model group metric by type. Maybe make a hash of metrics by task_type so it's easy to resolve in the continuation.

File: indico/types/metrics.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+from typing import List
```
Comment (nkassis) [inline indico/types/metrics.py:1]: Similarly I think we may want to either submodule this to model_groups or add model_group in the name of the file for clarity

File: indico/queries/metrics.py (hunk 1 above)
Comment (jkgenser) [reply indico/queries/metrics.py]: done.

File: indico/types/metrics.py (hunk 3 above)
Comment (jkgenser) [reply indico/types/metrics.py]: done.

File: indico/queries/metrics.py (hunk 2 above)
Comment (jkgenser) [reply indico/queries/metrics.py]: Added the GetModelGroupMetrics query in the continuation style as prescribed

**Comment by jkgenser** [comment]
Added retrain for metrics field. Client tests pass against dev after merging in refactored fog Evaluation schema.

File: indico/queries/model_groups/metrics.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        }
+      }
+    }
+    """
+
+    def __init__(self, model_group_id: int):
+        super().__init__(self.query, variables={"modelGroupId": model_group_id})
+
+    def process_response(self, response):
+        return json.loads(
```
Comment (nkassis) [inline indico/queries/model_groups/metrics.py:101]: There no object for this one?

Comment (jkgenser) [reply indico/queries/model_groups/metrics.py:101]: The object detection metrics are JSONString in the graphql schema

**Comment by jkgenser** [comment]
Rebased

**Comment by jkgenser** [comment]
> I think we should change the object detection metrics to a real schema (while keeping backwards for now) so we can make this interface nicer.

Will make JIRA for updating in graphql and then client.

**Pull request merged**